# VPN代理
宏基因组在线分析工具均为国外网站，使用国外的VPN会提高数据的传输效率

ExpressVPN

登录[主页](https://www.express-vpn.space/users/sign_in)下载安装客户端

> 登录账号：jingzhejiang@hotmail.com
> 
> 登录密码：问我

尝试使用分析网站所在区域的VPN代理，进行数据的上传及分析

# 主要网站

## [VIROME](http://virome.dbi.udel.edu/)：2012, 49 cited
### 比对的数据库
1. [UniRef clusters: a comprehensive and scalable alternative for improving sequence similarity searches](http://bioinformatics.oxfordjournals.org/content/31/6/926.full)
2. [MetagenomesOnline - The Curated Database for Environmental Metagenome Proteins](http://metagenomesonline.org/)

> 用户名：jingzhejiang
> 
> 密码：问我

==拼接后使用，因为长度低于300bp的序列该系统中不予考虑==

The VIROME pipeline analyzes ==only those read(s) > 300 bases in length==. In your dataset 0.0% of the bases are in the 0 read(s) > 300 bases in length. Based on this information and the graphic above, do you want to proceed with the analysis of this library?

Due to the excessive computational demands and poor information return from of analyzing single Illumina or Ion torrent reads, we ==require that Illumina or Ion torrent data be assembled prior to submission.== Unassembled Illumina or Ion torrent data is not suitable for the VIROME pipeline. 

Due to length filterthere is no data to process via VIROME analysis pipeline.Please ensure you have selected the correct file to uploadand/or have ==enough sequences over 300 bases for unassebmleddata or 600 bases for assembled data==

## [MetaVir](http://metavir-meb.univ-bpclermont.fr/)：2014, 42 cited
### 比对的数据库
BLAST comparison of the predicted proteins with [ncbi Refseq complete viral genomes proteins](ftp://ftp.ncbi.nih.gov/refseq/release/viral/)

> 用户名：jingzhejiang
> 
> 密码：问我

**由于空间受限，上传数据库功能暂时被关闭！**

## [EBI Metagenomics](https://www.ebi.ac.uk/metagenomics/)：2015, 6 cited
类似MetaVir的分析网站

The service identifies rRNA sequences, using rRNASelector, and performs taxonomic analysis upon 16S rRNAs using Qiime. The remaining reads are submitted for functional analysis of predicted protein coding sequences using the InterPro sequence analysis resource. InterPro uses diagnostic models to classify sequences into families and to predict the presence of functionally important domains and sites. By utilising this resource, the service offers a powerful and sophisticated alternative to BLAST-based functional metagenomic analyses. 

==上传数据后需要等几天才能看到==

> 注册邮箱：jingzhejiang@hotmail.com
> 
> 密码：问我

## [MG-RAST](http://metagenomics.anl.gov/): 2008, 1544 cited
[介绍](http://www.mcs.anl.gov/project/mg-rast-metagenomics-rast-server)

需要安装[firefox](https://www.mozilla.org/zh-CN/firefox/new/)浏览器

## [Taxonomer](https://www.taxonomer.com/)：2016, 人病原诊断网站

[全文链接](http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0969-1)

==这个网站简直是牛掰的不要不要的！==

> 注册邮箱：jingzhejiang@hotmail.com
>
> 密码：问我

**先把数据上传到SRA Database，然后再到Taxonomer中调用，可以节省分析时间**

### 文章摘抄

Taxonomer operates **at speeds comparable to the fastest tools**, Kraken [30] (~4 million reads/min) and CLARK [31] (~32 million reads/min), which provide only some of Taxonomer’s functionality. Unlike Kraken and CLARK, Taxonomer supports integrated nucleotide and protein-based classification for detection of highly diverse viral sequences at 10–100 times faster speeds than alignment-based tools with similar functionality (e.g. those used by SURPI [32]).

Taxonomer使用的是kmer的方法：Nucleotide-level classification in Taxonomer is based on exact k-mer matching. Taxonomer uses databases that are optimized for rapid k-mer queries that store every reference in which a k-mer is found as well as an associated k-mer weight for every reference. 

![image](https://static-content.springer.com/image/art%3A10.1186%2Fs13059-016-0969-1/MediaObjects/13059_2016_969_Fig1_HTML.gif)

SURPI软件使用的是alignment的方法，先用宿主基因组比对掉宿主基因，然后拿剩余的序列去比对数据库，寻找微生物序列。
**Taxonomer的结果与SURPI结果一致性较好**

### Taxonomer的分析结果

可以提供3种结果文件下载：

1. json：一种标准数据格式，可以被多种软件使用
2. xlsx：内容与json一样，更适合人阅读
3. 原始比对文件：类似blast结果文件，文件较大，提供有reads编号等共6列具体信息（多数只有4列信息）：

```
1. Bin category
2. Read classified (C) or unclassified (U) following bin assigment
3. Read name
4. Taxid of read assignment. For virus this corresponds to NCBI taxid. For bacteria, this corresponds to an artificial ID to the greengenes database, likewise for fungi except to the UNITE database. For Human this is an artificial ID to a gene.
5. Number of taxonomic levels at which the read assignment was made (more levels is a deeper assignment). Note this value cannot be compared between bin categories because difference taxonomies are used — for example, one cannot use this to compare the depth of a bacterial assignment to that of a viral assignment.
6. Read length
```

> 疑问：为什么很多reads的长度都为0？？？

### 使用说明

1. Quick Analysis提供快速分析，统计最多200k个reads
2. Full Analysis要求先上传全部文件后再慢慢分析
3. Ambiguous reads是指被注释为2或以上个物种的reads
4. 转为人类设计，if your particular organism is not present in our database, the reads derived from that organism may find homology with some other organism that is in our database. 结果分析时要考虑。


## [Pathosphere](http://www.pathosphere.org)：2015，病原检测网站

[全文链接](http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0840-5)

> 注册邮箱：jingzhejiang@hotmail.com
> 
> 密码：问我

[User's Guide](https://pathosphere.org/documents/10506/93d690d0-2308-4511-8c56-a94888012a72)

上传速度较快！分析内容很多！

## [insideDNA](https://insidedna.me/)