# 基本概念
1. The assembly of metagenomic sequence data
and the identification of the operational taxonomic units
(OTUs) are major technical challenges that persist in metagenomic data analysis pipelines
2. 拼接与不拼接对宏基因组多样性结果影响不大（liu et al. BMC Genomics 2011；Treangen et al. Gen Biol 2014）。可以用taxonomer做快速的分析。 it's becoming better to work with the reads rather than assembly - as they get longer and as newer algorithms are designed for longer reads
3. 拼接后相当于数据压缩，分析的速度会快一些。但拼接后还可以做基因功能、新基因组挖掘等其它分析。
4. 基因分析的工作相对基因组分析更容易、准确
5. 最重要的两点：reads大小、覆盖深度。
6. taxonomy分析：Metaphlan、Kraken、humann2（[及该实验室的其它资源](https://bitbucket.org/biobakery/biobakery/wiki/Home)）
7. **拼接结果验证**：https://www.cbcb.umd.edu/software/valet；==[ALE](https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bts723)==；iMetaAmos（集成在MetaAmos中）

# Commercialized Genomic Analysis Software：
[参考文献](http://bib.oxfordjournals.org/content/early/2014/09/01/bib.bbu030.long)
1. [CLC Genomics Workbench](https://www.qiagenbioinformatics.com/products/clc-genomics-workbench/)，25000元
2. [Geneious](http://www.geneious.com/)
3. [DNAStar](http://www.dnastar.com/t-allproducts.aspx)

# Alignment tools
BWA（Burrows-Wheeler Aligner）, Bowtie2, BLAST, BLAT, FASTM, RAPSearch, DIAMOND, [LAST](http://last.cbrc.jp/), Ublast

# Mapping Reads Back To Assembly Contigs
There is one main consideration in trying to reproduce the **coverage profile** from an assembly of short reads:

1. You do it during assembly without using information, or
2. You do it post-assembly, possibly losing information.

Most assemblers for NGS data won't do it by default during assembly since it takes up more memory to keep track of the reads pileup during the assembly process. Then the option is to do it post-assembly, which is not perfect. What mainly makes it imperfect is that one will **use a different tool** to ==pileup the reads back to the contigs== in a way that won't reproduce with 100% fidelity the decisions taken by the assembler during the process. **Repetitive regions** are one of the main sources of difference in this post-assembly pileup differences. Maybe other people know of other genomic features that will make this different.

[SciLifeLab Workshop](http://metagenomics-workshop.readthedocs.io/en/latest/functional-annotation/read_mapping.html)

[INF-BIO9120_fall2013](https://github.com/lexnederbragt/INF-BIO9120_fall2013_de_novo_assembly/blob/master/practicals/02_Mapping_reads_to_an_assembly.md)

[Simple fool's guide](http://sfg.stanford.edu/mapping.html)

[==多种工具比较==](http://merenlab.org/2015/06/23/comparing-different-mapping-software/)

http://merenlab.org/software/anvio/


## BWA（Burrows-Wheeler Aligner）

## Bowtie2


## [BBMap](http://sourceforge.net/projects/bbmap/) and [Samtools](http://www.htslib.org/)
[参考](https://sourceforge.net/p/metawatt/wiki/Map%20reads%20to%20contigs/)

## [Tablet](https://ics.hutton.ac.uk/tablet/)
Use Tablet to visualize resulting mapping.

# Assembly Validation

## [MOCAT2](http://mocat.embl.de/index.html)：2012，61 cited
华大开发的pipeline，使用SOAP做拼接

## [Galaxy](https://galaxyproject.org/)

[Wiki](https://wiki.galaxyproject.org/FrontPage)

[应用实例](https://indexofire.gitbooks.io/notebook_of_analyzing_pathogen_ngs_data/content/others/galaxy.html)

# Binning tools
Metagenomic binning is an essential task in analyzing metagenomic sequence datasets. Arguably the most popular metagenomic binning approaches are alignment-based methods. Another completely different binning approach is based on genomic sequence composition. Codon usage, oligonucleotide frequencies and GC content often are distinct in different genomes. Short k-mers, contiguous nucleotide fragments with k basepairs, have been shown to be both efficient and effective for metagenomic binning.

However, convergent evolutionary char-acteristics, including horizontal gene transfer, may diminish the accuracy of binning, especially for composition-based approaches and for the study of those taxa that may not be well-represented by the training data.


> And another tool for binning I've had some luck with and is very easy to run, compared with something like ESOM, but hasn't been mentioned yet is MaxBin: http://sourceforge.net/projects/maxbin/
## ==Alignment-based methods==
These methods are not necessarily ideal for the identification of novel genomes, though theymay be used to identify phylogenetic nodes that contain putatively novel lineages. But, similarity based methods may provide higher annotation accuracy and resolu- tion compared to compositional binning.

### MEGAN

### HMM-GRASPx: 2016

### CARMA：2008, 239 cited
### PhymmBL：2009, 334 cited
### MetaPhlAn
### PhyloSift
### MetaCluster-TA：2014, 14 cited
### MetaCluster 5.0：2012, 68 cited
## ==Composition-based methods==
All three techniques, Kraken, LMAT, and CLARK, attempt to limit the k-mer search space by either finding the least common ancestor (LCA) k-mers or finding discriminatory k-mers that uniquely identify an organism at a given taxonomy level.
### [Kraken](http://ccb.jhu.edu/software/kraken/)：2014，197 cited
[Article](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-3-r46)

### [CLARK](http://clark.cs.ucr.edu//)：2015, 35 cited
[Article](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1419-2)

### [LMAT](https://sourceforge.net/projects/lmat/)：2013, 52 cited
[Article](http://bioinformatics.oxfordjournals.org/content/29/18/2253.full)

### [SMART](https://bitbucket.org/ayl/smart)：2016
[Article](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1159-6)

### TETRA：2004, 268 cited
案例1
http://www.nature.com/ncomms/2013/130916/ncomms3420/full/ncomms3420.html/
### Phylopathia：2007, 422 cited
### PhylopithiaS
### [ESOMs](https://github.com/tetramerFreqs/Binning)：2009, 231 cited
Emergent Self-Organizing Maps

### Two-tiered clustering
### Opal: 2016

### [Kaiju](http://kaiju.binf.ku.dk/)：2015，
Taxonomic classification programs achieve high speed by comparing genomic k-mers. We show in a genome exclusion benchmark that Kaiju classifies reads with higher sensitivity and similar precision compared with current k-mer-based classifiers, especially in genera that are underrepresented in reference databases. We also demonstrate that Kaiju classifies up to 10 times more reads in real metagenomes. Kaiju can process millions of reads per minute and can run on a standard PC.

### [MyTaxa](enve-omics.ce.gatech.edu/mytaxa/)：2014, 22 cited
一种新的bining方法：The distinguishing aspect of MyTaxa is that it employs all genes present in an unknown sequence as classifiers, weighting each gene based on its (predetermined) classifying power at a given taxonomic level and frequency of horizontal gene transfer. MyTaxa also implements a novel classification scheme based on the genome-aggregate average amino acid identity concept to determine the degree of novelty of sequences representing uncharacterized taxa, i.e. whether they represent novel species, genera or phyla.

# Assemble tools

## IDBA

## SPAdes

## [Megahit](https://github.com/voutcn/megahit)：2015, 41 cited

https://github.com/voutcn/megahit/wiki/An-example-of-real-assembly

http://ivory.idyll.org/blog/2014-how-good-is-megahit.html

http://omictools.com/metagenomic-assembly-category

# 分析工具

## [DIAMOND](https://github.com/bbuchfink/diamond)：2015, 93 cited
We recently published a new alignment tool called DIAMOND that is able to align short metagenomic sequencing reads against the NCBI-nr database at 20000 times the speed of BLASTX without loss of sensitivity

[评论](http://www.polarmicrobes.org/?p=1068)

### [AC-DIAMOND](http://link.springer.com/chapter/10.1007/978-3-319-31744-1_38)
#### [链接网址](https://sourceforge.net/projects/ac-diamond/)
We introduce AC-DIAMOND, which attempts to speed up DIAMOND via better SIMD parallelization and reference indexing. Experimental results show that AC-DIAMOND is about 4 times faster than DIAMOND on aligning DNA reads or contigs while retaining the essentially the same sensitivity.
### [MEGAN](https://ab.inf.uni-tuebingen.de/software/megan6)：2011, 650 cited
[参考文献](http://www.ncbi.nlm.nih.gov/pubmed/27327495)

## [ViromeScan](http://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-2446-3): 2016, 2 cited
### 比对方法
The tool uses hierarchical databases for eukaryotic viruses to unambiguously assign reads to viral species more accurately and >1000 fold faster than other existing approaches.

## [HS-BLASTN](https://github.com/chenying2016/queries/tree/master/hs-blastn-src)：2015，
22 times faster than MegaBLAST and exhibits better parallel performance than MegaBLAST


## [综合网站](https://omictools.com/virome-category)
## [论坛](https://www.biostars.org/p/58279/)
## [网站评论](http://www.the-scientist.com/?articles.view/articleNo/40581/title/Metagenomics-Mash-Up/)
## Keywords
metagenom*, bioinformati*, tools, analysis, 


# 自主分析策略
## [参考文献1](http://www.pnas.org/content/112/38/11941.full)：
This latter database consisted of GenBank Viral Refseq entries plus all available complete viral genomes in National Center for Biotechnology Information (NCBI) and European Bioinformatics Institute and 512 predicted prophages identified using PhageFinder.
## 组装策略
1. 单个样品组装；
2. 同一家庭各成员样品混合一起，先分析Contigs >500 bp的，contigs与自己比较摘除冗余contigs；
3. 在第1步中未能被组装的raw reads再比对到contigs上；
4. 每个样品剩余的reads再做一次组装；
5. 全部contigs再合并一起，并去冗余；
6. 将全部reads再比对回contigs，筛掉嵌合体（contigs with a sudden drop in coverage or with positions where there were high numbers of partially mapped reads）；
7. 挑选出可以比对到contig末端的reads用于延长contig；
8. 用新工具再次拼接contig，并去冗余；
9. 再次用reads比对contigs，筛掉嵌合体；
10. 全部reads再次比对contigs，计算RPKM表达量。

## [参考文献2](http://www.nature.com/ismej/journal/v9/n11/full/ismej201548a.html)：
To identify SAG contigs originating from DNA viruses, we used a combination of five criteria, listed below.
1. Marker genes：(*phage, *virus, virion, prophage, terminase, capsid, head, tail, fiber, baseplate, portal, lysis, structural, T4, lambda, mu, lambdoid, podo*, myovir*, siphovir*, integrase, transposase).
2. DNA sequence anomalies：In bacterial genomes, a GC skew is associated with the origin of replication (Karlin and Campbell, 1998). GC skew can also be associated with the insertion of foreign DNA, including prophages (Grigoriev, 1998). Local anomalies in GC content and codon usage may also aid in the detection of prophages and other laterally acquired genetic elements within bacterial genomes (Akhter et al., 2012; Soares et al., 2012).
3. Metagenomic fragment recruitment：high metagenomic viral recruitment or low metagenomic bacterial recruitment

## 组装效果评估
![image](http://t1.qpic.cn/mblogpic/33053afa469140e41e34/460)
misoriented mate pairs, indicating a possible misjoin between unrelated genomic regions (a); a region with unusually deep coverage, indicating potential collapsed repeat (b); and a weak join, indicating a possible misjoin between unrelated genomic regions (c).

